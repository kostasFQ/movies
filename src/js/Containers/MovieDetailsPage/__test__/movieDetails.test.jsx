import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import ConnectedMovieDetailsPage from '../index';


const t1 = {
  startFetch: true,
  fetched: false,
  movie: {
    film: {vote_count: 4424,
    id: 278927,
    video: false,
    vote_average: 6.8,
    title: 'The Jungle Book',
    popularity: 12.258,
    poster_path: '/vOipe2myi26UDwP978hsYOrnUWC.jpg',
    original_language: 'en',
    original_title: 'The Jungle Book',
    genre_ids: [
      12,
      18,
      10751,
      14,
      16,
    ],
    backdrop_path: '/eIOTsGg9FCVrBc4r2nXaV61JF4F.jpg',
    adult: false,
    overview: 'A man-cub named Mowgli fostered by wolves. After a threat from the tiger Shere Khan, Mowgli is forced to flee the jungle, by which he embarks on a journey of self discovery with the help of the panther, Bagheera and the free-spirited bear, Baloo.',
    release_date: '2016-04-07'
  }
  },
};
const t2 = {
  startFetch: true,
  fetched: true,
  movie: {
    film: {vote_count: 4424,
    id: 278927,
    video: false,
    vote_average: 6.8,
    title: 'The Jungle Book',
    popularity: 12.258,
    poster_path: '/vOipe2myi26UDwP978hsYOrnUWC.jpg',
    original_language: 'en',
    original_title: 'The Jungle Book',
    genre_ids: [
      12,
      18,
      10751,
      14,
      16,
    ],
    backdrop_path: '/eIOTsGg9FCVrBc4r2nXaV61JF4F.jpg',
    adult: false,
    overview: 'A man-cub named Mowgli fostered by wolves. After a threat from the tiger Shere Khan, Mowgli is forced to flee the jungle, by which he embarks on a journey of self discovery with the help of the panther, Bagheera and the free-spirited bear, Baloo.',
    release_date: '2016-04-07'
  }
  },
};

const mockStore = configureStore();
const store1 = mockStore(t1);
const store2 = mockStore(t2);

test('test component 1', () => {
  const render = renderer.create(
    <ConnectedMovieDetailsPage store={store1} />,
  );

  const component = render.toJSON();
  expect(component).toMatchSnapshot();
});

test('test component 2', () => {
  const render = renderer.create(
    <ConnectedMovieDetailsPage store={store2} />,
  );

  const component = render.toJSON();
  expect(component).toMatchSnapshot();
});
