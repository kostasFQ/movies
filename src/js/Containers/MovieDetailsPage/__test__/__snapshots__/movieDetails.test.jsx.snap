// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1.test MOVIE DETAILS PAGE component render 1`] = `
<div>
  <div
    className="curtian"
  >
    <img
      alt="no poster"
      className="poster"
      src="http://image.tmdb.org/t/p/w500//VuukZLgaCrho2Ar8Scl9HtV3yD.jpg"
    />
  </div>
  <div
    className="container"
  >
    <div
      className="container"
    >
      <h1
        className="h1"
      >
        VENOM
      </h1>
      <ul
        className="list"
      >
        <li
          className="item"
        >
          Action
        </li>
        <li
          className="item"
        >
          Crime
        </li>
        <li
          className="item"
        >
          Horror
        </li>
        <li
          className="item"
        >
          Science Fiction
        </li>
      </ul>
      <div
        className="container"
      >
        <div
          className="whiteStars"
        />
        <div
          className="fillStars"
          style={
            Object {
              "width": "132px",
            }
          }
        />
        <div
          className="container"
        >
          6.6
        </div>
      </div>
    </div>
    <div
      className="buttonsBlock"
    >
      <button
        className="container"
        onClick={[Function]}
        type="button"
      >
        Watch Now
      </button>
      <button
        className="container"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
  </div>
</div>
`;

exports[`2.test MOVIE DETAILS PAGE component render 1`] = `
<div>
  <div
    className="curtian"
  >
    <img
      alt="no poster"
      className="poster"
      src="http://ctoetotakoe.ru/wp-content/uploads/2016/05/404-not-found.png"
    />
  </div>
</div>
`;

exports[`3.test MOVIE DETAILS PAGE component render 1`] = `
<div
  className="loader"
>
  Spiner
</div>
`;
